app/scheduler_func.py

```diff
@@ -179,3 +179,98 @@ def YQPoint_Distributions(request):
     else:
         dis_id = int(dis_id)
         return YQPoint_Distribution(request, dis_id)
+
+
+"""
+使用方式：
+
+scheduler.add_job(changeActivityStatus, "date", 
+    id=f"activity_{aid}_{to_status}", run_date, args)
+
+注意：
+    1、当 cur_status 不为 None 时，检查活动是否为给定状态
+    2、一个活动每一个目标状态最多保留一个定时任务
+
+允许的状态变换：
+    2、报名中 -> 等待中
+    3、等待中 -> 进行中
+    4、进行中 -> 已结束
+
+活动变更为进行中时，更新报名成功人员状态
+"""
+def changeActivityStatus(aid, cur_status, to_status):
+    try:
+        with transaction.atomic():
+            activity = Activity.objects.select_for_update().get(id=aid)
+            if cur_status is not None:
+                assert cur_status == activity.status
+            if cur_status == Activity.Status.APPLYING:
+                assert to_status == Activity.Status.WAITING
+            elif cur_status == Activity.Status.WAITING:
+                assert to_status == Activity.Status.PROGRESSING
+            elif cur_status == Activity.Status.PROGRESSING:
+                assert to_status == Activity.Status.END
+            else:
+                raise ValueError
+
+            activity.status = to_status
+    
+            if activity.status == Activity.Status.WAITING:
+                if activity.bidding:
+                    # TODO 元气值抽签
+                    pass
+
+                # 提醒参与者
+                scheduler.add_job(notifyActivity, 'date', id=f"activity_{activity.id}_remind",
+                    run_date=activity.act_start - timedelta(minutes=15), args=[activity.id, 'remind'], replace_existing=True)
+
+
+            # 活动变更为进行中时，修改参与人参与状态
+            if activity.status == Activity.Status.PROGRESSING:
+                participants = Participant.objects.filter(activity_id=aid, status=Participant.AttendStatus.APLLYSUCCESS)
+                for participant in participants:
+                    # TODO: 现在是都变成参加成功，暂时还没做签到
+                    participant.status = Participant.AttendStatus.ATTENDED
+                    participant.save()
+
+            activity.save()
+
+
+    except:
+        # TODO send message to admin to debug
+        pass
```

Comment:
别写成这样，至少写成
```suggestion
    except Exception as e:
        print(e)
```
这样还能在命令行里看到报错，不然测试的既不显示出错位置，也不显示出错类型，根本找不到问题

Reply:
好的